Updating Roulette odds for Roulette Remix

SuperAj3 opened the pull request:
Please pull the change authored on 8 June 2025 for the June Roulette Remix odds next Thursday 12 June.

SuperAj3 committed:
Update randomChaoItemList.go

Added Cartridge to pool and corrected RC Electromagnet odds

SuperAj3 committed:
Update randomChaoItemList.go

Roulette odds amended to display buddies and characters appearing in Group 1 of Roulette Remix (June 12 to July 11).

JKouga reviewed (changes requested):
Group 1 characters good.  The next group of buddies is incorrect

SuperAj3 committed:
Update randomChaoItemList.go

Added Marine Chao to odds list
NOTE - group 1 characters and Group 2 buddies.
Run Group 1 buddies with Group 2 characters in July

SuperAj3 committed:
Update randomChaoItemList.go

Added Death Egg to group 2 (J noted it was missing from all groups). Recalculated odds for non-rate up SRs to accommodate.

SuperAj3 committed:
Update randomChaoItemList.go

Added RC Piggybank which was missed from the odds.
